# HG changeset patch
# User Rik <rik@octave.org>
# Date 1445574891 25200
#      Thu Oct 22 21:34:51 2015 -0700
# Node ID 10704158315e8194add2c8f1be8767730097639d
# Parent  3e449f43e7080a2fa513380f9fcca30eadd4b60b
Plot errorbar correctly for just a single point (bug #46165).

* __errplot__.m: If calculated bar width is 0, substitute 0.1.

diff --git a/scripts/plot/draw/private/__errplot__.m b/scripts/plot/draw/private/__errplot__.m
--- a/scripts/plot/draw/private/__errplot__.m
+++ b/scripts/plot/draw/private/__errplot__.m
@@ -280,30 +280,36 @@ function h = __do_errplot__ (fstr, hax, 
 
 endfunction
 
 function [xdata, ydata] = errorbar_data (xdata, ydata, ldata, udata,
                                          xldata, xudata, ifmt,
                                          xscale, yscale)
   if (strcmp (xscale, "linear"))
     dx = 0.01 * (max (xdata(:)) - min (xdata(:)));
+    if (dx == 0)
+      dx = .1;
+    endif
     xlo = xdata - dx;
     xhi = xdata + dx;
   else
     n = xdata > 0;
     if (! any (n))
       n = xdata < 0;
     endif
     logdata = log (abs (xdata(n)));
     rx = exp (0.01 * (max (logdata) - min (logdata)));
     xlo = xdata/rx;
     xhi = xdata*rx;
   endif
   if (strcmp (yscale, "linear"))
     dy = 0.01 * (max (ydata(:)) - min (ydata(:)));
+    if (dy == 0)
+      dy = .1;
+    endif
     ylo = ydata - dy;
     yhi = ydata + dy;
   else
     n = ydata > 0;
     if (! any (n))
       n = ydata < 0;
     endif
     logdata = log (abs (ydata(n)));
